keyboards/skyloong/gk980/bl/ansi/keymaps/via/keymap.c

```diff
@@ -0,0 +1,75 @@
+        _______,  _______,  _______,  _______,   _______,  _______,     _______,   _______,   _______,  _______,         _______,                          _______,    BL_UP,     KC_P1,    KC_P2,   KC_P3,  KC_PENT,
+        KC_LCTL,  KC_LGUI,  KC_LALT,             _______,  _______,                _______,             KC_RALT,  MO(_WIN_INDEX),    KC_APP,               KC_LEFT,  BL_DOWN,  KC_RGHT,    KC_P0,  KC_PDOT
+    ),
+
+    [_MAC_INDEX] = LAYOUT_all(
+        _______,              KC_F1,    KC_F2,    KC_F3,    KC_F4,       KC_F5,     KC_F6,     KC_F7,    KC_F8,          KC_F9,     KC_F10,   KC_F11,   KC_F12,   G(KC_SPC),     KC_INS,  KC_HOME,  KC_END, KC_PAUS,
+        _______,  _______,  _______,  _______,  _______,  _______,    TO(_WIN),  TO(_MAC),   _______,  _______,        BL_TOGG,    BL_BRTG,  BL_BRTG,   KC_DEL,                  KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,   _______,   _______,  _______,        _______,    _______,  _______,  _______,                   KC_P7,    KC_P8,   KC_P9, KC_PPLS,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,   _______,   _______,  _______,        _______,    _______,            _______,                   KC_P4,    KC_P5,   KC_P6,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,   _______,   _______,  _______,        _______,                        _______,       BL_UP,      KC_P1,    KC_P2,   KC_P3, KC_PENT,
+        KC_LCTL,  KC_LALT,  KC_LGUI,            _______,  _______,                _______,             KC_RCMD, MO(_MAC_INDEX),    KC_RALT,            KC_LEFT,     BL_DOWN,    KC_RGHT,    KC_P0,  KC_PDOT
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_WIN] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(BL_DOWN, BL_UP), ENCODER_CCW_CW(C(KC_WH_D), C(KC_WH_U)), ENCODER_CCW_CW(G(C(KC_LEFT)), G(C(KC_RGHT)))},
+    [_MAC] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_F14, KC_F15), ENCODER_CCW_CW(G(KC_MINS), G(KC_EQL)), ENCODER_CCW_CW(C(KC_UP), C(KC_DOWN))},
+    [_WIN_INDEX] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(BL_DOWN, BL_UP), ENCODER_CCW_CW(C(KC_WH_D), C(KC_WH_U)), ENCODER_CCW_CW(G(C(KC_LEFT)), G(C(KC_RGHT)))},
+    [_MAC_INDEX] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_F14, KC_F15), ENCODER_CCW_CW(G(KC_MINS), G(KC_EQL)), ENCODER_CCW_CW(C(KC_UP), C(KC_DOWN))}
+};
+#endif
+
+layer_state_t default_layer_state_set_user(layer_state_t state) {
+
+  switch (get_highest_layer(state)) {
+    case 0:
+        gpio_write_pin_low(MAC_PIN);
+        break;
+    case 1:
+        gpio_write_pin_high(MAC_PIN);
+        break;
+    default:
+        gpio_write_pin_low(MAC_PIN);
+
+    }
```

Comment:
This entire block could be written as: 
```suggestion
    gpio_write_pin(MAC_PIN, layer_state_cmp(state, 1));
```